packages/astro/src/core/create-vite.ts

```diff
@@ -71,6 +71,7 @@ export async function createVite(
 			astroIntegrationsContainerPlugin({ config: astroConfig }),
 		],
 		publicDir: fileURLToPath(astroConfig.publicDir),
+		base: astroConfig.base,
```

Comment:
This lets Vite's dev server take care of the subpath in `public` assets, i.e. `<img src="/subpath/images/penguin.png" />` still resolves in the `public` directory